File: python/p3/p3.py
# Problem 3
#
# The prime factors of 13195 are 5, 7, 13 and 29.
# What is the largest prime factor of the number 600851475143?

num = 600851475143
prime = 2

while num > 1:
    if num % prime == 0:
        num /= prime
    else:
        prime += 1

print prime

File: csharp/P1_to_10/P3.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using ProjectEuler.Helpers;

namespace ProjectEuler.P1_to_10
{
    /// <summary>
    /// https://projecteuler.net/problem=3
    /// </summary>
    internal class P3
    {
        private const long Number = 600851475143;

        /// <summary>
        /// Returns the largest prime factor of the number 600851475143
        /// </summary>
        public int Solve()
        {
            return PrimeHelpers.PrimeFactors(Number).Max();
        }
    }
}
